index.js

```diff
@@ -0,0 +1,237 @@
+    this.dry_run = dry_run === true
+    this.owner = owner
+    this.pr_id = pr_id
+    this.repo = repo
+    this.token = token
+
+    if (!this.token) {
+      const err = new Error('Missing github token. Set GITHUB_TOKEN env var.')
+      err.code = 'EMISSINGTOKEN'
+      throw err
+    }
+
+    if (!this.dry_run) {
+      if (!this.pr_id) {
+        const err = new Error('Missing pr_id. Please specify the PR ID')
+        err.code = 'EMISSINGPRID'
+        throw err
+      }
+    }
+
+    this.client = new Octokit({
+      auth: token
+    })
+
+    this.old_pct = 0
+    this.new_pct = 0
+    this.changes = new Map()
+  }
+
+  async run() {
+    console.log('running...')
+    await this.compare()
+    console.log('done', {
+      'old': this.old_pct
+    , 'new': this.new_pct
+    })
+    if (!this.dry_run) {
+      await this.hideOldComments()
+      await this.createPRComment()
+    }
+
+    // Reset
+    const changes = this.changes
+    const old_pct = this.old_pct
+    const new_pct = this.new_pct
+    this.changes = new Map()
+    this.old_pct = 0
+    this.new_pct = 0
+    return {
+      old_pct
+    , new_pct
+    , changes
+    }
+  }
+
+  async hideOldComments() {
+    const comments_list = await this.client.graphql(GET_ISSUES_QUERY, {
+      owner: this.owner
+    , repo: this.repo
+    , issue_id: this.pr_id
+    })
+
+    const comments = comments_list.repository.pullRequest.comments.nodes
+
+    for (const issue of comments) {
+      if (issue.isMinimized || !issue.body.includes(COMMENT_HEADER)) continue
+      console.log('minimize issue %s', issue.id)
+      await this.client.graphql(HIDE_COMMENTS_QUERY, {
+        owner: this.owner
+      , repo: this.repo
+      , id: issue.id
+      , classifier: 'OUTDATED'
+      })
+    }
+  }
+
+  async createPRComment() {
+    const body = `${COMMENT_HEADER}\n\n${this.buildIssueContent()}`
+
+    await this.client.issues.createComment({
+      owner: this.owner
+    , repo: this.repo
+    , issue_number: this.pr_id
+    , body
+    })
+  }
+
+  async getLatestCoverage() {
+    // Fetch the latest release from GitHub.
+    // If the release has a coverage.json file.
+    const release = await this.client.repos.getLatestRelease({
+      owner: this.owner
+    , repo: this.repo
+    })
+
+    let asset_id
+
+    for (const asset of release.data.assets) {
+      if (asset.name === 'coverage-summary.json') {
+        asset_id = asset.id
+        break
+      }
+    }
+
+    if (!asset_id) {
+      const err = new Error('Unable to find previous coverage file')
+      err.code = 'ENOPREVCOVERAGE'
+      throw err
+    }
+
+    const asset = await this.client.repos.getReleaseAsset({
+      owner: this.owner
+    , repo: this.repo
+    , asset_id
+    , request: {
+        headers: {
+          accept: 'application/octet-stream'
+        , authorization: `token ${this.token}`
+        }
+      }
+    })
+
+    const latest = JSON.parse(Buffer.from(asset.data))
```

Comment:
So this is an octect-stream coming down the pipe?  I didn't know JSON.parse could take a buffer like this.  Doesn't it have to fully download it first?  I don't fully grasp what's going on here.

Reply:
JSON.parse() on a buffer implicitly calls toString() on the buffer. It has already been downloaded at this point. `asset.data` is the actual buffer. I actually should be able to remove the `Buffer.from()` call here though since it's already a buffer. Doing that now